Automattic/sensei
# Add editor wizard modal for courses and lessons

Fixes #5125

### Changes proposed in this Pull Request

* It removes `_needs_template` logic (old logic to add template when created through Course Outline).
* It removes Learning Mode onboarding.
* It adds a modal when creating new Courses and Lessons (it also applies to lessons created dynamically through the Course Outline).
* This PR doesn't add styles to the modal. I'm leaving it to the "steps" issues.
* It refactors the message we show to the user when adding/removing Sensei blocks to the post editor. Discussion: p1652822680825379-slack-C07418EJ0
* The Sensei Pro product tour will also be removed as part of another PR in Sensei Pro.

### Testing instructions


* Clear the Local Storage.
* If you're using a more recent version of Gutenberg, remove user meta `wp_persisted_preferences` ([more details](https://github.com/WordPress/gutenberg/commit/d7c18ee5ea1f1b2014e66d14f56b00a16f697caf)).
* Create a new Course, and a new Lesson through the "New Course" and "New Lesson" buttons in their respective sections under WP Admin. Make sure you see a modal when creating them. Also, make sure that they don't conflict with the Gutenberg blocks modal from the core (our modal is just displayed after closing the Gutenberg modal).
* Close the modal, save the post, and refresh the page. Make sure the modal doesn't appear again.
* Now, create a lesson through the Course Outline.
* Edit the lesson, and make sure the modal is displayed.
* Close the modal, and make sure it adds the default blocks template to the Lesson.
* Save the post, refresh the page, and make sure the modal isn't displayed again.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/168915458-cbbc6f1a-0658-41b9-baf7-e8c8c91e22d9.mp4
